test/e2e/fixtures.go

```diff
@@ -474,3 +476,27 @@ func createTestBusyboxPods(tb testing.TB, data *TestData, num int, ns string, no
 
 	return podNames, podIPs, cleanupFn
 }
+
+// testServiceFromNode is a helper function which uses polling mechanism
+// to make sure that Service has been realized.
+func testServiceFromNode(t *testing.T, url, nodeName string, timeout time.Duration, statusReady bool) error {
+	cmd := fmt.Sprintf("curl -s --connect-timeout 1 --retry-connrefused %s", url)
+	err := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		t.Logf(cmd)
+		rc, stdout, stderr, err := RunCommandOnNode(nodeName, cmd)
+		if rc == 0 && err == nil && statusReady == true {
+			return true, nil
+		} else if rc != 0 && statusReady == false {
+			// Service no longer accessible.
+			return true, nil
+		} else if rc == 0 && err == nil && statusReady == false {
+			// keep trying, Service still accessible.
+			return false, nil
+		} else if err != nil {
+			t.Logf("stdout: %s - stderr: %s - err: %v", stdout, stderr, err)
+		}
```

Comment:
it's rare to use three conditions, it's error-prone, Could you simplify this? maybe put them into two cases like below:
```
switch statusReady {
case true:
case false:
}
```

Reply:
done, thanks.